components/sup/src/config.rs

```diff
@@ -190,7 +192,7 @@ mod tests {
         use super::*;
         #[test]
         fn local_addr_for_gossip_listen_addr_works_for_unspecified_address() {
-            let listen_addr = GossipListenAddr::default();
+            let listen_addr = GossipListenAddr::configured_value();
```

Comment:
Using `default` in tests is OK, and the safer thing to do. I'd only use `configured_value()` if you're explicitly testing that behavior (e.g., that it responds to the right environment variable, etc.) 

By using it here _without_ explicit need to, it creates uncertainty as to what is actually being tested ("Did this test fail because I have a certain environment variable set to something the test wasn't expecting?"). It also makes the test subject to interference from other tests that _do_ rely on the configured environment variable that might be running at the same time (gotta use the `locked_env_var!` macro for that!)